lib/internal/abort_controller.js

```diff
@@ -185,25 +190,69 @@ class AbortSignal extends EventTarget {
     return signal;
   }
 
+  /**
+   * @param {AbortSignal[]} signals
+   * @returns {AbortSignal}
+   */
+  static any(signals) {
+    validateAbortSignalArray(signals, 'signals');
+    const resultSignal = createAbortSignal({ composite: true });
+    const resultSignalWeakRef = new WeakRef(resultSignal);
+    for (let i = 0; i < signals.length; i++) {
+      const signal = signals[i];
+      if (signal.aborted) {
+        abortSignal(resultSignal, signal.reason);
+        return resultSignal;
+      }
+      resultSignal[kSourceSignals] ??= new SafeSet();
+      signal[kDependantSignals] ??= new SafeSet();
+      if (!signal[kComposite]) {
+        resultSignal[kSourceSignals].add(new WeakRef(signal));
+        signal[kDependantSignals].add(resultSignalWeakRef);
+      } else {
+        if (!signal[kSourceSignals]) {
```

Comment:
This if branch can be removed by moving it as an else if to the previous if statement. This would improve the branch prediction for jit.